Epsilon.Abstractions/Components/IComponentManager.cs

```diff
@@ -0,0 +1,6 @@
+namespace Epsilon.Abstractions.Components;
+
+public interface IComponentManager
+{
+    public Task<IWordCompetenceComponent> Fetch(int courseId, string pageName, string userId, string domainName);
```

Comment:
First of all, you cannot be sure that all components that are returned have word support unless we explicitly decide that all components should have Word support for them to be components. The problem is that we wanted to support multiple export formats in the beginning, which would not be possible with this setup.

So I suggest a choice:
- Enforce Word as an export format for all components, and keep it inheritely limited to Word.
- Use a generic parameter in the componentloader class, and return that in the fetch function. Afterwhich you can check if the returned component supports certain export functions depending on which is selected.

Secondly (which is also why I despise this interface), courseId and pageName are only used by pages. Their structure is much different from all the other components, so I suggest to make them different from components **or** remove this interface, because it doesn't add much in terms of functionality.

Last but not least, domainName should be handled elsewhere, you should just simply give the fully populated domain to this loader as a type. Currently it violates SOLID principles, namely the single responsibility principle, and it will cause performance issues if every component has to retrieve the same data over and over again.

Reply:
Ill try to make it generic

Reply:
I've completely removed the interface, making it possible to use different file types is still not solved, will refactor that in a different pull request